Scripts/Data/Type/SurfaceData.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace VRTK.Core.Data.Type
+{
+    using UnityEngine;
+
+    /// <summary>
+    /// The SurfaceData contains information about the located surface.
+    /// </summary>
+    public class SurfaceData : TransformData
+    {
+        /// <summary>
+        /// The origin in which the surface search came from.
+        /// </summary>
+        public Vector3 origin;
+        /// <summary>
+        /// The direction in which the surface search came from.
+        /// </summary>
+        public Vector3 direction;
+        /// <summary>
+        /// RaycastHit data about the current collision.
+        /// </summary>
+        public RaycastHit CollisionData
+        {
+            get
+            {
+                return _collisionData;
+            }
+            set
+            {
+                PreviousCollisionData = _collisionData;
+                _collisionData = value;
+            }
+        }
+        /// <summary>
+        /// RaycastHit data about the previous collision.
+        /// </summary>
+        public RaycastHit PreviousCollisionData
+        {
+            get;
+            protected set;
+        }
+
+        protected RaycastHit _collisionData;
+
+        /// <summary>
+        /// Default Constructor.

```

Comment:
Maybe this should use the same documentation the other constructors get?

`Creates a new SurfaceData with a default origin and direction for the collision search.` would be OK if we apply the other requested change on the other constructor.